docs/src/main/sphinx/security/jwt-auth.rst

```diff
@@ -0,0 +1,145 @@
+secure Trino with a custom single sign-on or proxy solution. If that describes
+you, then enable JWT authentication by setting the :doc:`JWT authentication type
+<authentication-types>` in ``etc/config.properties``:
+
+.. code-block:: properties
+
+    http-server.authentication.type=JWT
+
+In addition, you must configure the ``http-server.authentication.jwt.key-file``
+property in ``etc/config.properties``. For example:
+
+.. code-block:: text
+
+    http-server.authentication.jwt.key-file=https://cluster.example.net/.well-known/jwks.json
+
+The following configuration properties for ``etc/config.properties`` are
+available:
+
+.. list-table:: Configuration properties for JWT authentication
+   :widths: 50 50
+   :header-rows: 1
+
+   * - Property
+     - Description
+   * - ``http-server.authentication.jwt.key-file``
+     - Required. Specifies either the URL to a JWKS service or the path to a
+       PEM or HMAC file, as described below this table.
+   * - ``http-server.authentication.jwt.required-issuer``
+     - Optional. Specifies a string that must match the value of the JWT's
+       ``iss`` field to consider this JWT valid for this Trino server.
+       The ``iss`` field in the JWT identifies the principal that issued the
+       JWT.
+   * - ``http-server.authentication.jwt.required-audience``
+     - Optional. Specifies a string that must match the value of the JWT's
+       ``aud`` field to consider this JWT valid for this Trino server.
+       The ``aud`` field in the JWT identifies the recipients that the JWT is
+       intended for.
+   * - ``http-server.authentication.jwt.principal-field``
+     - Optional. String to identify the field in the JWT that identifies the
+       subject of the JWT. The default value is ``sub``.
+   * - ``http-server.authentication.jwt.user-mapping.pattern``
+     - Optional. A regular expression pattern to map all user names for this
+       authentication system to the format expected by the Trino server. See
+       :doc:`User mapping </security/user-mapping>`.
+   * - ``http-server.authentication.jwt.user-mapping.file``
+     - Optional. The path to a JSON file that contains a set of user mapping
+       rules for this authentication system. See :doc:`User
+       mapping </security/user-mapping>`.
+
+Use the ``http-server.authentication.jwt.key-file`` property to specify
+either:
+
+-  The URL to a JWKS endpoint service, where the URL begins with ``https://``
+   and ends with the port number. The JWKS service must be reachable from the
+   coordinator. If the coordinator's local network locks down port access for
+   security, then administrators must either open a port to the JWKS, or look in
+   the server's ``/.well-known/jwks.json`` file. `RFC 5785
+   <https://tools.ietf.org/html/rfc5785>`_ describes the ``.well-known``
+   mechanism.
+
+   .. caution::
+
+        The Trino server also accepts JWKS URLs that begin with ``http://``, but
```

Comment:
I assume this renders and indents properly

Reply:
Yep. Indented within the bulleted list, as intended.